src/transformers/models/megatron_bert/convert_megatron_bert_checkpoint.py

```diff
@@ -0,0 +1,268 @@
+            output_state_dict[self_attn_name + ".value.weight"] = v
+            output_state_dict[self_attn_name + ".value.bias"] = v_bias
+
+            # Clear the stored tensor.
+            attention_qkv_weight = None
+
+        # Copy weights and biases as is.
+        elif weight_or_bias in ["weight", "bias"]:
+
+            out_name = megatron_to_transformers[op_name]
+            output_state_dict[layer_name + out_name + weight_or_bias] = val
+
+    # The final layernorm.
+    output_state_dict["bert.encoder.ln.weight"] = transformer["final_layernorm.weight"]
+    output_state_dict["bert.encoder.ln.bias"] = transformer["final_layernorm.bias"]
+
+    # The config.
+    output_config = {
+        "vocab_size": word_embeddings.size(0),
+        "hidden_size": 1024,
+        "num_hidden_layers": 24,
+        "num_attention_heads": 16,
+        "hidden_act": "gelu_new",
+        "intermediate_size": 4096,
+        "hidden_dropout_prob": 0.1,
+        "attention_probs_dropout_prob": 0.1,
+        "max_position_embeddings": 512,
+        "type_vocab_size": 2,
+        "initializer_range": 0.2,
+        "layer_norm_eps": 1e-12,
+        "gradient_checkpointing": False,
+        "position_embedding_type": "absolute",
+        "use_cache": False,
+    }
+
+    # The pooler.
+    pooler = lm["pooler"]
+
+    # Store the matrix and the bias.
+    output_state_dict["bert.pooler.dense.weight"] = pooler["dense.weight"]
+    output_state_dict["bert.pooler.dense.bias"] = pooler["dense.bias"]
+
+    # The LM head from Megatron (for RACE).
+    lm_head = model["lm_head"]
+
+    # The transform matrix.
+    output_state_dict["cls.predictions.transform.dense.weight"] = lm_head["dense.weight"]
+    output_state_dict["cls.predictions.transform.dense.bias"] = lm_head["dense.bias"]
+
+    # The transform LN.
+    output_state_dict["cls.predictions.transform.LayerNorm.weight"] = lm_head["layernorm.weight"]
+    output_state_dict["cls.predictions.transform.LayerNorm.bias"] = lm_head["layernorm.bias"]
+
+    # For the decoder, we replicate the weights.
+    output_state_dict["cls.predictions.decoder.weight"] = word_embeddings
+    output_state_dict["cls.predictions.bias"] = lm_head["bias"]
+
+    # The classifier from Megatron (for MLNI).
+    binary_head = model["binary_head"]
+
+    # Store the classifier.
+    output_state_dict["cls.seq_relationship.weight"] = binary_head["weight"]
+    output_state_dict["cls.seq_relationship.bias"] = binary_head["bias"]
+
+    # It should be done!
+    return output_state_dict, output_config
+
+
+####################################################################################################
+
+
+def main():
+    # Create the argument parser.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--print-checkpoint-structure", action="store_true")
+    parser.add_argument("path_to_checkpoint", type=str, help="Path to the ZIP file containing the checkpoint")
```

Comment:
Should this have the `--` prefix?
```suggestion
    parser.add_argument("--path_to_checkpoint", type=str, help="Path to the ZIP file containing the checkpoint")
```

Reply:
If we add the `--` it changes the command invocation from `python3 convert.py <path_to_file>` to `python3 convert.py --path_to_checkpoint <path_to_file>`. That's extra typing. Also, the `<path_to_file>` argument is mandatory so the positional argument is more common in that context. I would vote against adding the `--` (but I can do it if you think it's needed/more coherent with the rest of Transformers).